Fix artist name

gorgobacka opened the pull request:
At the moment, whipper always uses 'Artist in MusicBrainz' for track artist and album artist. This can lead for example to '[unknown]' in filenames.

This PR fix this issue (also mentioned here: https://github.com/JoeLametta/whipper/issues/155) and uses the artist-credit -> name if it exists (and therefor the name is different from 'Artist name in MusicBrainz'), and otherwise artist-credit -> artist -> name.

Freso commented:
It would be great with a test case to both show case the issue and also document that your fix is working.

gorgobacka commented:
@Freso Thanks a lot for your comment. I will try to include a test case over the weekend.

gorgobacka commented:
I rebased the code unto latest master and added a test case to demonstrate that the metadata is set correctly for both situations. However, the test does not check the filenames.

@Freso Do you think a test to check the filenames is also needed? I expect it's not, since the variable artist (instead of sort-name) is used in the filenames.

Freso commented on whipper/test/test_common_mbngs.py:
```diff
@@ -114,3 +114,40 @@ def testMalaInCuba(self):
+
+    def testNorthernGateway(self):
+        # disc with artists tagged with [unknown] and an alias in MusicBrainz
+        # see https://github.com/JoeLametta/whipper/issues/155
```
Make this a [docstring](http://www.pythonforbeginners.com/basics/python-docstrings/) instead of a comment. :)

gorgobacka replied:
Thanks for your suggestions. I will change it soon.

JoeLametta replied:
Looking forward to merge it when it's completed.

gorgobacka commented:
I expect my previous changing needs some explanation. During my recent tests, I detected that sometimes the artist name in the recoding is different from the artist name on the release. I thought this also fits to this PR and added a fix for it. Before, whipper used always the artist name of the recording. However, this is not always correct. Now it uses the artist name of the release first. 

@Freso Can you give some insights about the artists named in recordings, please? Do you think, they are handled correctly now? What about track titles? I expect they can be also different (this might be a separate PR).

JoeLametta commented:
Let's hear what @Freso thinks about it: my knowledge regarding to MusicBrainz's inner details is quite limited.

Freso commented:
A Track title can be different than a Recording title (and IMHO, whipper should use the Track titles).

In the same vein, I feel whipper should use the Track artist credit and disregard the Recording artist credit entirely. (Though possibly as a switch for classical - in MusicBrainz, classical releases have the composer as the Track artist and the performers as the Recording artist(s).)

JoeLametta commented:
@gorgobacka Hi, Freso replied. What's your status update about this pull request?

gorgobacka commented:
@JoeLametta It can be merged. The Track Title can be addressed in a different PR. Same for a switch for classical releases.

Note: At the moment I'm still using the Recording artist as fallback, if no Track artist is present in the json file.

RecursiveForest commented:
This looks good to me, although I can't say I know much about the semantics of MusicBrainz entries. I'm happy to have this merged to fix issue 155 and to address using track title & using a classical switch in another PR. If I don't hear a NAK from @JoeLametta, @Freso, or @Wizzup in the next few days I'll press the button.

gorgobacka commented:
@RecursiveForest, @JoeLametta If there is nothing missing, can you please merge this PR?

JoeLametta commented:
> If there is nothing missing, can you please merge this PR?

Merged, thanks.

gorgobacka commented:
Thanks.
